internal/hud/server/server.go

```diff
@@ -360,9 +360,10 @@ func (s *HeadsUpServer) HandleNewSnapshot(w http.ResponseWriter, req *http.Reque
 }
 
 func (s *HeadsUpServer) GetView(ctx context.Context, req *proto_webview.GetViewRequest) (*proto_webview.View, error) {
-	return &proto_webview.View{
-		FatalError: "AHHH FATAL ERROR",
-	}, nil
+	state := s.store.RLockState()
```

Comment:
Are you missing an unlock?

Reply:
I am! This explains the hanging behavior I was seeing!